src/Control/ElementFormController.php

```diff
@@ -55,7 +56,19 @@ public function finished()
         $user->finished();
 
         $page = $this->getPage();
-        $controller = Injector::inst()->create($page->getControllerName(), $this->element->getPage());
+
+        if ($page === null) {
+            return null;
+        }
+
+        while (!$page instanceof SiteTree) {
+            $page = $page->getPage();
```

Comment:
What if the `DataObject` doesn't have a `getPage()` method? It's totally valid to have an `ElementalArea` on an arbitrary `DataObject` with no link to `SiteTree` at all.

Reply:
In one of our project we have added -https://github.com/dnadesign/silverstripe-elemental-list and because of that we have elemental form added inside of another elemental block due to which it was not giving the Page Controller, i.e. I am checking for the instanceof SiteTree....

Reply:
Ahh okay. In that case we should be checking explicitly whether this `$page` is an instance of `BaseElement`. I'll request that change.

Reply:
sure, have updated it now to this